packages/version/src/__tests__/version-command.spec.ts

```diff
@@ -1009,4 +1009,20 @@ describe('VersionCommand', () => {
       expect((collectUpdates as Mock).mock.calls[0][3].isIndependent).toBe(true);
     });
   });
+
+  describe('CLI arguments', () => {
+    it('throws when --dry-run has a string argument other than "true" or "false"', async () => {
+      const cwd = await initFixture('lifecycle');
+      const command = lernaVersion(cwd)('--no-changelog', '--dry-run', 'premajor');
+
+      await expect(command).rejects.toThrow('--dry-run option must be of type boolean and you provided: premajor');
+    });
+
+    it('does not throw when --dry-run has a string argument "true"', async () => {
```

Comment:
I'm not sure why you say that, I'm pretty sure that we can pass "true" or "false" to any yargs and it should be parsed because the problem we had was that passing "premajor" was parsed as `true` because the result of `Boolean("premajor")` is `true` and that is what we want to avoid here. The `coerce` I've put in place will only allow a boolean or the string "string" or "false", anything else will throw which I think is correct

Reply:
> I'm not sure why you say that, I'm pretty sure that we can pass "true" or "false" to any yargs and it should be parsed because the problem we had was that passing "premajor" was parsed as `true` because the result of `Boolean("premajor")` is `true` and that is what we want to avoid here. The `coerce` I've put in place will only allow a boolean or the string "string" or "false", anything else will throw which I think is correct.

I checked exactly how `yargs` parsed the command line, see #576.

Yargs does not try to coerce `premajor` into a boolean. That is what I was trying to show with my screen shots in #569.